sql/src/test/java/org/apache/druid/sql/calcite/CalciteQueryTest.java

```diff
@@ -5269,44 +5267,6 @@ public void testSqlIsNullToInFilter()
     );
   }
 
-  @Test
-  public void testInFilterWith23Elements()
```

Comment:
shouldn't this test be kept?
I think if it gets removed the subq transformed codepath is not covered by the other tests

Reply:
`org.apache.druid.sql.calcite.CalciteQueryTest#testPlanWithInFilterLessThanInSubQueryThreshold` and `org.apache.druid.sql.calcite.CalciteJoinQueryTest#testPlanWithInFilterMoreThanInSubQueryThreshold` should cover both sides of this. 
The comment in this ticket implies that it was for a particular issue when druid did not support joins. Is there a reason to keep it?